pages/profile.tsx

```diff
@@ -0,0 +1,43 @@
+import { useUser } from '../lib/hooks'
+import Layout from '../components/Layout'
+import ButtonPrimary from '../components/ButtonPrimary';
+
+const Profile = ({user}) => {
+    return (
+        <Layout title="User Profile" user={user}>
+          <div style={{margin: "3em"}}>
+            <h1>Profile</h1>
+            {user && (
+                <>
+                  <div>Username: {user.username}</div>
+                  <div>Created: {new Date(Number(user.created)).toISOString()}</div>
+                  <div>Roles: <ul>{user.roles.map(role => <li key={role}>{role}</li>)}</ul></div>
+                </>
+            )}
+            {user && user.roles.includes('admin') && (<a href="/roles">Role Management</a>)}
+            <ButtonPrimary href="/api/logout">Logout</ButtonPrimary>
+          </div>
+        </Layout>
+    )
+}
+
+import { loginRequired } from '../lib/loggedin'
```

Comment:
it's a good idea to keep all imports at the top of the file

Reply:
Moved import to top of file